src/main/java/org/wso2/lsp4intellij/IntellijLanguageClient.java

```diff
@@ -342,4 +343,12 @@ public static void removeWrapper(LanguageServerWrapper wrapper) {
     public static Map<String, Set<LanguageServerWrapper>> getProjectToLanguageWrappers() {
         return projectToLanguageWrappers;
     }
+
+    public static void didChangeConfiguration(DidChangeConfigurationParams params,
```

Comment:
Let's avoid depending on LSP packages (`org.eclipse.lsp4j`) here. We can create `DidChangeConfigurationParams` inside the `RequestManager`. Is it correct to assume the intellij-plugin will invoke this method whenever there's config change?

Reply:
@rasika As I understood this is directly exposed as an user API so that users can dynamically trigger the `didChangeConfiguration()` (@gayanper correct me if I'm wrong). So to make things aligned, IMO this should be exposed via `IntelliJLanguageClient`.

Anyway since @rasika 's concern is also valid, we can implement this method in `RequestManager` and just call it here, just like what I've done with the timeouts API? WDYT?

Reply:
@NipunaRanasinghe we still need to have the `IntellijLanguageClient.didChangeConfiguration()`. My concern was about `DidChangeConfigurationParams`. `IntellijLanguageClient` doesn't need to aware `org.eclipse.lsp4j.*` classes.  Since we already have used these classes in other places, please ignore my comment :)

Reply:
@rasika do you mean to add the DidChangeConfigurationParams inside the RequestManager class as a inner class ? IMO its good to have it as a sibling class ?

Reply:
@NipunaRanasinghe `didChangeConfiguration` was something i added additionally since its related to my requirement. But the original change here is to listen for server messages such as `ResponseMessage`. The `InitializeResult` is one kind of result that we will get from ResponseMessage.

Reply:
@rasika got your point :) +1 for keeping this as it is.

Reply:
@NipunaRanasinghe do i need to change anything more on this ?

Reply:
@gayanper  IMO you dont have to :)